src/xdg.rs

```diff
@@ -262,6 +262,63 @@ impl IconLoader {
         Self { icons }
     }
 
+    fn premultiply_generic(data: &mut [u8]) {
+        // TODO: change to array_chunks_mut() once that is stabilised.
+        for chunk in data.chunks_exact_mut(4) {
+            if let [r, g, b, a] = chunk {
+                let r = *r as u16 * *a as u16 + 127;
+                let g = *g as u16 * *a as u16 + 127;
+                let b = *b as u16 * *a as u16 + 127;
+                chunk[0] = ((r + (r >> 8) + 1) >> 8) as u8;
+                chunk[1] = ((g + (g >> 8) + 1) >> 8) as u8;
+                chunk[2] = ((b + (b >> 8) + 1) >> 8) as u8;
+            }
+        }
+    }
+
+    #[cfg(all(target_arch = "aarch64", target_feature = "neon"))]
+    unsafe fn premultiply(data: &mut [u8]) {
+        use core::arch::aarch64::*;
+
+        let one = vdupq_n_u16(1);
+        let half = vdupq_n_u16(127);
+
+        // Work on “just” 8 pixels at once, since we need the full 16-bytes of
+        // the q registers for the multiplication.
+        //
+        // TODO: change to array_chunks_mut() once that is stabilised.
+        let mut iter = data.chunks_exact_mut(8 * 4);
+        while let Some(chunk) = iter.next() {
+            let chunk = chunk.as_mut_ptr();
+            let uint8x8x4_t(mut r8, mut g8, mut b8, a8) = vld4_u8(chunk);
+
+            // This is the same algorithm as the other premultiply(), but on
+            // packed 16-bit instead of float.
+            let mut r16 = vmull_u8(r8, a8);
```

Comment:
```suggestion
            // This is the same algorithm as the other premultiply(), but on
            // packed 16-bit instead of float.
            
            let mut r16 = vmull_u8(r8, a8);
```

Space to help indicate it's not just about the statement following the comment directly.